lib/improver/tests/nbhood/square_kernel/test_SquareNeighbourhood.py

```diff
@@ -667,44 +667,46 @@ def test_with_masked_data(self):
         neighbourhood processing when the input cubes contain masked
         arrays."""
         expected_data = np.array(
-            [[1.66666667, 1.66666667, -1.22222222, -1.22222222, -1.22222222,
-              -1.33333333, 1.66666667],
-             [1.66666667, 1.66666667, -1.22222222, -1.22222222, -1.22222222,
-              -1.33333333, 2.44444444],
-             [3.22222222, 4., 0.88888889, 0.88888889, 0.88888889,
-              1., -1.22222222],
-             [-1.22222222, -1.22222222, 0.88888889, 0.88888889, 0.88888889,
-              1., -1.22222222],
-             [-1.22222222, -1.22222222, 0.88888889, 0.88888889, 0.88888889,
-              1., -1.22222222],
-             [-1.22222222, -1.22222222, 1., 1., 1.,
-              1., -2.11111111],
-             [-2.88888889, -3.66666667, -1.22222222, -1.22222222, -1.22222222,
-              -1.33333333, 1.66666667]])
+            [[0.66666667, 0.33333333, -0.55555556, -0.66666667, -0.33333333,
```

Comment:
These numbers don't make a lot of sense to me, but the units tests you've modified below from the end of the process do.

Reply:
Please of the speed up method the data in the halo is meaningless rubbish (the arrays are rolled and wrapped around so the halo data is not a correct calculation). It is only the data in the middle which has valid data and I've checked these in Excel and these are the right values.